### DISCUSSION TITLE
How install Grott on RPi running HAOS

### DISCUSSION THREAD

**Discussion by rowanbradley** [johanmeijer/grott#532] [Q&A]
I've failed to get the growatt integration to work so need to try grott. Can I install grott on the same RPi that's running HA under HAOS? How? Please give blow by blow instructions (I'm a beginner at this...).

Thank you - Rowan

**Comment by PedroKTFC** [comment]
Yes you can. You need to go to Settings>Add-ons. Once there, in the bottom right click on ADD-ON STORE (blue button). Once there, in the top right hand corner there's a 3 dots item, click on that which gives you a menu that includes Repositories. Ckick on that and a window pops up. You need to add the following repository (copy and paste it into the Add field):  https://github.com/egguy/grott-home-assistant-add-on

Click on ADD to the right.

This gives you a new repository which includes a few Grott options. Click on Grott Beta branch (2.8) and install it. You then need to select the relevant options for your set up (mainly the inverter type).

Good luck!

**Comment by typxxi** [comment]
is there any alternative to run the grott code inside HA OS starting that from the terminal ?

My grott is running now a pi 3 while my ha is running on a pi5 which means i have a lot of power I could use to also run the grott installation .

I use the grott ha integration provided by muppet 3000 and that is working great but I would like to run the grott application on my ha server to reduce the amount of power needed and avoid another risk.

**Reply by PedroKTFC** [reply]
Do you mean grott or grottserver? If you run grott as an add-on, it runs under HA OS. That's how I run it.

**Comment by typxxi** [comment]
> Do you mean grott or grottserver? If you run grott as an add-on, it runs under HA OS. That's how I run it.

Ok, thanks for clarifying, but my question is,
if that addon is **THE ONLY WAY** to run grott under HA OS 

OR 

is there another way for example to install grott manualy via terminal ?

**Reply by PedroKTFC** [reply]
I don't know I'm afraid. I like running it as an add-on as configuring it is easy and I can start and stop it at will (I usually run grottserver but sometimes I stop it and go back to the Growatt servers to make some changes to my system so I need to repoint grott).

**Reply by egguy** [reply]
The design of HAOS dosen't lend itself very well to running custom service. The OS is built around docker.
I have seen some people running grottserver with the add-on `ssh and termail` but it is not really planned for this nor reproducible.

There is also mention of portainer, which would allows you to run custom container.

The  "proper" way to have a custom service running on HAOS is to create an add-on: https://developers.home-assistant.io/docs/add-ons/tutorial/ this would allow running your service, with monitoring and restart from HA.

**Reply by PedroKTFC** [reply]
I run grottserver using the "Advanced SSH & Web Terminal" add-on with the command `python3 grottserver.py 1> /dev/null &` and it happily sits there in the background. I have to run my own version as I couldn't work out how to load libscrc so I replaced it with local code.

**Comment by typxxi** [comment]
> The "proper" way to have a custom service running on HAOS is to create an add-on: https://developers.home-assistant.io/docs/add-ons/tutorial/ this would allow running your service, with monitoring and restart from HA.

OK, then I will install your addon now and hope to migrage my first inverter from the dedicated grott pi-3 to your grott addon on my Pi-5 and hope for the best.

Fingers crossed.

thanks a lot.

**Reply by egguy** [reply]
If you want to manage the installation yourself, and not depend on my addon, is just a folder with a docker file inside.

You can create an addons folder as in the tutorial and drop the content of one of my addon inside (e.g. https://github.com/egguy/addon-grott-beta/tree/main/grott-beta) then you can build/edit/modify the container from within HA.

**Reply by typxxi** [reply]
No, I do not want such docker installation inside HA when that is not supported officially cause one fine day it might break even though it had been working for years and then I have to find out that it was my strange idea to run a docker. 

I have tried your grott add on for HA OS, installed it, started it, then moved a first inverter and then your addon forwared the messages to the grott cloud. 
Meanwhile a part of your addon stopped which was not happy with my mapping where I had mentioned an right now unsupported MIC inverter (even though that had worked out on my grott pi 3 installation). I then had changed the "MiC" into "MIN" and your addon kept on working and treating my mic like a MIN. 

But unfortunatelly the sensors of this inverter created by grott integration (muppet 3000) are still no longer updated anymore.

I wrote this down here just for those who might find this topic and can read along what i have achieved and where i am right now, while I also have written more details in another topic under your project as an open issue.

So far the installation went well when I got the right access point what repository I had to add. 
It really went on smoothly till I reached the point where the system did not like my maybe unsupported MIC inverter which might not work right now. But that is a different story I guess or I might move on and try to move a min inverter first, but not today.
thanks

Here is the current state of my "grott addon installation" 
https://github.com/egguy/addon-grott-beta/issues/31#issuecomment-1998516811

**Reply by egguy** [reply]
> No, I do not want such docker installation inside HA when that is not supported officially cause one fine day it might break even though it had been working for years and then I have to find out that it was my strange idea to run a docker.

This is the official way and how add-on are run. The add-on is just the compiled docker image run by the HA docker daemon. The `addons` folder is intended to run your own addon without the need to publish.
